torch/distributed/pipelining/_IR.py

```diff
@@ -796,17 +811,21 @@ def move_param_to_callee(
                     else ""
                 )
             )
+
+            # Get submodule
             callee = root.get_submodule(callee_name)
             assert not hasattr(
                 callee, param_fqn
             ), f"Module {callee_name} already has a parameter named {param_fqn}"
+
+            # Assign the parameter to the submodule
             if is_buffer:
                 _assign_attr(
```

Comment:
im kinda confused though, how come we can assign the attr to a submodule and not cause fqn duplication?

Reply:
We are **moving** the attr to the submodule. 
The original attr will be removed IIRC.